virtual_rainforest/models/plants/community.py

```diff
@@ -0,0 +1,121 @@
+"""The :mod:`~virtual_rainforest.models.plants.community` submodule provides a simple
+dataclass to hold plant cohort information and then the ``Plants`` class that is used to
+hold list of plant cohorts by grid cell and generate those lists from variables provided
+in the data object.
+
+NOTE - much of this will be outsourced to pyrealm.
+
+"""  # noqa: D205, D415
+
+
+from dataclasses import dataclass
+
+from virtual_rainforest.core.data import Data
+from virtual_rainforest.core.logger import LOGGER
+from virtual_rainforest.models.plants.functional_types import (
+    PlantFunctionalType,
+    PlantFunctionalTypes,
+)
+
+
+@dataclass
+class PlantCohort:
+    """A dataclass describing a plant cohort.
+
+    The cohort is defined by the plant functional type, the number of individuals in the
+    cohort and the diameter at breast height for the cohort.
+    """
+
+    pft: PlantFunctionalType
+    dbh: float
+    n: int
+
+
+class Plants:
+    """A dictionary of plant cohorts keyed by grid cell id.
+
+    An instance of this class is initialised from a
+    :class:`~virtual_rainforest.core.data.Data` object that must contain the variables
+    ``plant_cohort_cell_id``, ``plant_cohort_pft``, ``plant_cohort_n`` and
+    ``plant_cohort_dbh``. These are required to be equal length, one-dimensional arrays
+    that provide the data to initialise each plant cohort. The data are validated and
+    then compiled into lists of cohorts keyed by grid cell id. The class provides a
+    __getitem__ method to allow the list of cohorts for a grid cell to be accessed using
+    ``plants_inst[cell_id]``.
+
+    Args:
+        data: A data instance containing the required plant cohort data.
+        pfts: The plant functional types to be used.
+    """
+
+    def __init__(self, data: Data, pfts: PlantFunctionalTypes):
+        self.data: dict = dict()
```

Comment:
Find it confusing that this is called `data` and then `data` is also an argument, but that the two aren't the same thing?

Reply:
Also a fair point!

Reply:
Changed to `communities`